From e44bdd2046648c968685f8a254a2f0dbb602275d Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Sat, 6 Aug 2022 11:35:55 +0200
Subject: [PATCH] Fix Huawei empty string of ANDROID_STORAGE

Huawei hi6250 define in init.hi6250.rc ANDROID_STORAGE to "", so check empty string and replace with default path. Apply change for all directory
---
 core/java/android/os/Environment.java | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/core/java/android/os/Environment.java b/core/java/android/os/Environment.java
index 9e13f05129a2..9dec6ed85675 100644
--- a/core/java/android/os/Environment.java
+++ b/core/java/android/os/Environment.java
@@ -1364,9 +1364,13 @@ public class Environment {
         return !defaultScopedStorage && !forceEnableScopedStorage;
     }
 
+    /**
+     * Huawei hi6250 define in init.hi6250.rc ANDROID_STORAGE to "", so check empty string and replace with
+     * default path. Apply change for all directory
+     */
     static File getDirectory(String variableName, String defaultPath) {
         String path = System.getenv(variableName);
-        return path == null ? new File(defaultPath) : new File(path);
+        return ((path == null || path.isEmpty()) ? new File(defaultPath) : new File(path));
     }
 
     /** {@hide} */
-- 
2.30.2

